pkg/controllers/statusaggregator/controller.go

```diff
@@ -20,192 +19,229 @@ package statusaggregator
+	fedcorev1a1informers "github.com/kubewharf/kubeadmiral/pkg/client/informers/externalversions/core/v1alpha1"
 	"github.com/kubewharf/kubeadmiral/pkg/controllers/statusaggregator/plugins"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util/delayingdeliver"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util/eventsink"
 	"github.com/kubewharf/kubeadmiral/pkg/controllers/util/propagationstatus"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util/sourcefeedback"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util/worker"
 	"github.com/kubewharf/kubeadmiral/pkg/stats"
+	clusterutil "github.com/kubewharf/kubeadmiral/pkg/util/cluster"
+	"github.com/kubewharf/kubeadmiral/pkg/util/eventhandlers"
+	"github.com/kubewharf/kubeadmiral/pkg/util/eventsink"
+	"github.com/kubewharf/kubeadmiral/pkg/util/fedobjectadapters"
+	"github.com/kubewharf/kubeadmiral/pkg/util/informermanager"
+	"github.com/kubewharf/kubeadmiral/pkg/util/logging"
+	"github.com/kubewharf/kubeadmiral/pkg/util/naming"
+	"github.com/kubewharf/kubeadmiral/pkg/util/worker"
 )
 
 const (
-	ControllerName = "status-aggregator-controller"
+	StatusAggregatorControllerName = "status-aggregator-controller"
 
-	EventReasonUpdateSourceObjectStatus     = "UpdateSourceObjectStatus"
-	EventReasonUpdateSourceObjectAnnotation = "UpdateSourceObjectAnnotation"
+	EventReasonUpdateSourceObjectStatus = "UpdateSourceObjectStatus"
 )
 
 // StatusAggregator aggregates statuses of target objects in member clusters to status of source object
 type StatusAggregator struct {
-	// name of the controller: <sourceKind>-status-aggregator
 	name string
 
-	// Store for the federated type
-	federatedStore cache.Store
-	// Controller for the federated type
-	federatedController cache.Controller
-	// Client for federated type
-	federatedClient util.ResourceClient
-
-	// Store for the source type
-	sourceStore cache.Store
-	// Controller for the source type
-	sourceController cache.Controller
-	// Client for source type
-	sourceClient util.ResourceClient
-
-	// Informer for resources in member clusters
-	informer util.FederatedInformer
 	// For triggering reconciliation of all target resources. This is
 	// used when a new cluster becomes available.
 	clusterQueue            workqueue.DelayingInterface
 	clusterAvailableDelay   time.Duration
 	clusterUnavailableDelay time.Duration
 	objectEnqueueDelay      time.Duration
 
-	worker        worker.ReconcileWorker
-	typeConfig    *fedcorev1a1.FederatedTypeConfig
+	worker        worker.ReconcileWorker[reconcileKey]
 	eventRecorder record.EventRecorder
 
-	// plugin for source type to aggregate statuses
-	plugin  plugins.Plugin
 	metrics stats.Metrics
 	logger  klog.Logger
+
+	dynamicClient dynamicclient.Interface
+	fedClient     fedclient.Interface
+
+	federatedInformer              informermanager.FederatedInformerManager
+	federatedObjectInformer        fedcorev1a1informers.FederatedObjectInformer
```

Comment:
We name the informers as `fedObjectInformer` and `clusterFedObjectInformer` in other controllers, should we keep the consistent naming style?